swift/xcode-autobuilder/XcodeProjectParser.cpp

```diff
@@ -217,41 +233,29 @@ static std::vector<fs::path> collectFiles(const std::string& workingDir) {
       continue;
     }
     if (p.extension() == ".xcodeproj" || p.extension() == ".xcworkspace") {
-      files.push_back(p);
+      structure.xcodeFiles.push_back(p);
     }
   }
-  return files;
+  return structure;
 }
 
 static std::unordered_map<std::string, std::vector<std::string>> collectWorkspaces(
-    const std::string& workingDir,
-    bool& swiftPackageEncountered) {
+    const ProjectFiles& projectFiles) {
   // Here we are collecting list of all workspaces and Xcode projects corresponding to them
   // Projects without workspaces go into the same "empty-workspace" bucket
-  swiftPackageEncountered = false;
   std::unordered_map<std::string, std::vector<std::string>> workspaces;
   std::unordered_set<std::string> projectsBelongingToWorkspace;
-  std::vector<fs::path> files = collectFiles(workingDir);
-  for (auto& path : files) {
+  for (auto& path : projectFiles.xcodeFiles) {
     if (path.extension() == ".xcworkspace") {
       auto projects = readProjectsFromWorkspace(path.string());
       for (auto& project : projects) {
         projectsBelongingToWorkspace.insert(project.string());
         workspaces[path.string()].push_back(project.string());
       }
-    } else if (!swiftPackageEncountered && path.filename() == "Package.swift") {
-      // a package manifest must begin with a specific header comment
-      // see https://docs.swift.org/package-manager/PackageDescription/PackageDescription.html
-      static constexpr std::string_view packageHeader = "// swift-tools-version:";
-      std::array<char, packageHeader.size()> buffer{};
-      std::string_view bufferView{buffer.data(), buffer.size()};
-      if (std::ifstream{path}.read(buffer.data(), buffer.size()) && bufferView == packageHeader) {
-        swiftPackageEncountered = true;
-      }
     }
   }
   // Collect all projects not belonging to any workspace into a separate empty bucket
-  for (auto& path : files) {
+  for (auto& path : projectFiles.xcodeFiles) {
     if (path.extension() == ".xcodeproj") {
       if (projectsBelongingToWorkspace.count(path.string())) {
```

Comment:
```suggestion
      if (projectsBelongingToWorkspace.contains(path.string())) {
```
thank you C++20 :wink: